### PR TITLE
Branch 22.10 fix mg bfs type mismatch

### LINKED ISSUE #2515: [BUG] Dask-cudf dataframes no longer merges dataframes of different type
**Describe the bug**
A recent change in `cudf` no longer allows inner merges of dataframes with different type and this is breaking `MG BFS`. In fact, when checking if the user is passing valid seeds to `MG BFS`, an inner merge between all the vertices in the graph and the `BFS seeds` is computed but this fails if the dataframes are of different type. In such case, `cudf` can return unexpected results as described in the error message below
```
UserWarning: Merging dataframes with merge column data type mismatches:
+---------------+------------+-------------+
| Merge columns |left dtype  | right dtype |
+---------------+------------+-------------+
|    (0, 0)     |   int32    |  int64      |
+---------------+------------+-------------+
Cast dtypes explicitly to avoid unexpected results.
```
**Steps/Code to reproduce bug**
Run `BFS` with seeds of different type than the graph vertices's type

### PR DESCRIPTION
**Pull request by jnke2016**
This PR coerces the seeds and vertices type, raise a warning and updates the tests accordingly. 
closes #2515.

### CODE REVIEW

File: python/cugraph/cugraph/dask/traversal/bfs.py [hunk 1]
```diff
@@ -122,12 +122,28 @@ def bfs(input_graph,
 
     if not isinstance(start, (dask_cudf.DataFrame, dask_cudf.Series)):
         if not isinstance(start, (cudf.DataFrame, cudf.Series)):
-            start = cudf.Series(start)
+            vertex_dtype = input_graph.nodes().dtype
+            start = cudf.Series(start, dtype=vertex_dtype)
         if isinstance(start, (cudf.DataFrame, cudf.Series)):
             # convert into a dask_cudf
             start = dask_cudf.from_cudf(start, input_graph._npartitions)
```
Comment (rlratzel) [inline python/cugraph/cugraph/dask/traversal/bfs.py:129]: Seems like we didn't need the `if isinstance` block, or am I missing something?
```suggestion
        # convert into a dask_cudf
        start = dask_cudf.from_cudf(start, input_graph._npartitions)
```

File: python/cugraph/cugraph/dask/traversal/bfs.py [hunk 2]
```diff
@@ -122,12 +122,28 @@ def bfs(input_graph,
...
 
     def check_valid_vertex(G, start):
```
Comment (rlratzel) [inline python/cugraph/cugraph/dask/traversal/bfs.py:131]: It's not obvious why we need to define a function here just to call it once as part of the `if check_start` block below.  Should this code just be moved to the `if check_start` block?

File: python/cugraph/cugraph/tests/mg/test_mg_bfs.py [hunk 3]
```diff
@@ -126,6 +126,11 @@ def test_dask_bfs_invalid_start(dask_client, directed):
     with pytest.raises(ValueError):
         dcg.bfs(G, source_vertex).compute()
 
+    # invalid dtype
```
Comment (rlratzel) [inline python/cugraph/cugraph/tests/mg/test_mg_bfs.py:129]:
```suggestion
    # invalid dtype (the default cudf.Series() dtype is int64)
```

File: python/cugraph/cugraph/dask/traversal/bfs.py (hunk 1 above)
Comment (jnke2016) [reply python/cugraph/cugraph/dask/traversal/bfs.py:129]: Right. I am missing and `else` statement here which should raise a type error.

```
else:
     raise TypeError("'start' must be a list or a cudf object or a dask 
                     f"object, got :  {type(start)}")
```

Comment (jnke2016) [reply python/cugraph/cugraph/dask/traversal/bfs.py:129]: Right. This extra check is unnecessary

**Comment by rlratzel** [comment]
rerun tests

Rerunning due to a conda server error.

**Review by rlratzel — APPROVED** [review]

**Pull request merged**